src/vs/base/parts/quickinput/browser/quickInput.ts

```diff
@@ -328,6 +352,13 @@ class QuickInput extends Disposable implements IQuickInput {
 		this.ui.ignoreFocusOut = this.ignoreFocusOut;
 		this.ui.setEnabled(this.enabled);
 		this.ui.setContextKey(this.contextKey);
+
+		const validationMessage = this.validationMessage || this.noValidationMessage;
+		if (this._lastValidationMessage !== validationMessage) {
+			this._lastValidationMessage = validationMessage;
+			dom.reset(this.ui.message, ...renderLabelWithIcons(escape(validationMessage)));
+			this.showMessageDecoration(this.severity);
```

Comment:
I'll just move this to a separate `if` to ensure it updates when set after the validation message.